[feature] 리뷰 추가 기능을 개선한다.

hek-git opened the pull request:
## #️⃣ 연관된 이슈
> ex) #이슈번호, #이슈번호
#68
## 📝 작업 내용

> 이번 PR에서 작업한 내용을 간략히 설명해주세요(이미지 첨부 가능)
- 테크 스펙 : (테크 스펙 url)
- 각 Entity에 존재하는 UUID 필드 타입을 모두 id(Long 타입)로 통일합니다.
- 리뷰 추가 요청시에 메뉴에 대해서도 리뷰할 수 있도록 메뉴 리스트 필드를 추가하였습니다.
  -  ReviewMenu command, domain, Entity 객체 등 추가하였습니다.
  - 이와 관련하여 각각 Service, Repository 클래스에서 적절히 변환하여 추가하도록 하였습니다.
- common-domain 모듈에 구현되어있는 util 클래스 사용하여 uuid 파싱 관련 중복 코드를 줄였습니다.

### 스크린샷 (선택)
**메뉴에 대한 리뷰가 있는 리뷰 추가 요청**
![스크린샷 2024-08-15 오후 3 42 53](https://github.com/user-attachments/assets/d2e63e19-c165-46fe-9ae2-df7a27bc3a39)   
   
   
**메뉴에 대한 리뷰가 없는 리뷰 추가 요청**
![스크린샷 2024-08-15 오후 3 44 38](https://github.com/user-attachments/assets/a559e4b5-80b3-4e9b-b20c-d971ff9468a1)


## 💬 리뷰 요구사항(선택)

> 리뷰어가 특별히 봐주었으면 하는 부분이 있다면 작성해주세요
>
> ex) 메서드 XXX의 이름을 더 잘 짓고 싶은데 혹시 좋은 명칭이 있을까요?

---
### RCA 룰
| 코드 리뷰 멘토가 PR작성자에게 어떤 의도로 전달 되길 바라는지 알파벳으로 표현해주세요!
- r(request change): 꼭 반영
- c(comment): 왠만하면 반영
- a(approve): 반영해도 좋고 넘어가도 좋은 의견

jcw1031 committed:
[fix] gradle 멀티 모듈 의존성 정리 및 bootJar 옵션 수정

- gradle 빌드 시 mainClass를 찾지 못해 오류가 발생했습니다.
- runnable한 모듈인 mono 모듈만 bootJar.enable을 true로 설정했습니다.
- 기존 중복되었던 의존성을 정리했습니다.

hek-git committed:
[chore] menu-domain 패키지 의존성 추가
- menuRepository를 사용하기 위한 menu-domain 의존성 추가

hek-git committed:
[chore] menu-domain 패키지 의존성 추가
- menuRepository를 사용하기 위한 menu-domain 의존성 추가

hek-git committed:
[feature] ReviewMenuCreateRequest 클래스 추가
- 특정 메뉴에 대해서도 리뷰가 가능하도록 클래스 추가
  - menuUUid, content, isLiked 필드를 가짐

hek-git committed:
[feature] ReviewMenuCreateRequest 클래스 추가
- 특정 메뉴에 대해서도 리뷰가 가능하도록 클래스 추가
  - menuUUid, content, isLiked 필드를 가짐

hek-git committed:
[feature] ReviewCreateRequest 클래스 필드 추가
- 리뷰 생성 요청시 메뉴에 대해서도 리뷰하는 menus 필드를 추가한다. 이때 menus 필드가 없을 시에는 빈 리스트를 넣는다.

hek-git committed:
[feature] ReviewCreateRequest 클래스 필드 추가
- 리뷰 생성 요청시 메뉴에 대해서도 리뷰하는 menus 필드를 추가한다. 이때 menus 필드가 없을 시에는 빈 리스트를 넣는다.

hek-git committed:
[feature] ReviewCreateResult 클래스 추가
- 리뷰 추가 요청에 대한 결과를 반환하는 Result 클래스 추가
   - reivewUuid와 reviewMenuUuid 리스트를 갖는다.

hek-git committed:
[feature] ReviewCreateResult 클래스 추가
- 리뷰 추가 요청에 대한 결과를 반환하는 Result 클래스 추가
   - reivewUuid와 reviewMenuUuid 리스트를 갖는다.

hek-git committed:
[feature] ReviewMenuCreateCommand 클래스 추가
- menuUuid, content, isRecommend 필드 포함
- 변환 메서드 포함

hek-git committed:
[feature] ReviewMenuCreateCommand 클래스 추가
- menuUuid, content, isRecommend 필드 포함
- 변환 메서드 포함

hek-git committed:
[feature] ReviewMenu 도메인 클래스 추가
- Entity 생성에 필요한 필드 포함

hek-git committed:
[feature] ReviewMenu 도메인 클래스 추가
- Entity 생성에 필요한 필드 포함

hek-git committed:
[feature] ReviewMenuEntity 저장하기 위한 ReviewMenuEntityRepository 인터페이스 추가
- JPARepository 상속

hek-git committed:
[feature] ReviewMenuEntity 저장하기 위한 ReviewMenuEntityRepository 인터페이스 추가
- JPARepository 상속

hek-git committed:
[feature] ReviewMenuEntity 저장하기 위한 ReviewMenuRepository 인터페이스 추가
- ReviewMenu 리스트를 저장하는 saveAll 메서드 추가
- Domain -> Entity로 변환하는 toEntity 메서드 추가
- Entity -> Domain로 변환하는 toDomain 메서드 추가

hek-git committed:
[feature] ReviewMenuEntity 저장하기 위한 ReviewMenuRepository 인터페이스 추가
- ReviewMenu 리스트를 저장하는 saveAll 메서드 추가
- Domain -> Entity로 변환하는 toEntity 메서드 추가
- Entity -> Domain로 변환하는 toDomain 메서드 추가

hek-git committed:
[feature] ReviewCreateCommand 클래스 필드 추가
- ReviewMenuCreateCommand 리스트 필드 추가
- 관련 변환 메서드 수정

hek-git committed:
[feature] ReviewCreateCommand 클래스 필드 추가
- ReviewMenuCreateCommand 리스트 필드 추가
- 관련 변환 메서드 수정

hek-git committed:
[feature] 리뷰 요청에 대한  응답을 반환하는 ReviewCreateResponse 객체 필드 추가
- reviewMenuId 리스트 필드 추가
- 변환 메서드 추가

hek-git committed:
[feature] 리뷰 요청에 대한  응답을 반환하는 ReviewCreateResponse 객체 필드 추가
- reviewMenuId 리스트 필드 추가
- 변환 메서드 추가

hek-git committed:
[feature] Review 도메인 객체 필드 추가
- id 필드 추가
- restaurantUuid -> restaurantId로 변경

hek-git committed:
[feature] Review 도메인 객체 필드 추가
- id 필드 추가
- restaurantUuid -> restaurantId로 변경

hek-git committed:
[feature] ReviewEntity 클래스 필드 변경
- Restaurant의 pk를 갖도록함
- createdAt, updatedAt 필드 추가

hek-git committed:
[feature] ReviewEntity 클래스 필드 변경
- Restaurant의 pk를 갖도록함
- createdAt, updatedAt 필드 추가

hek-git committed:
[feature] ReviewMenuEntity 클래스 필드 변경
- Byte 형태 저장을 위한 UUID 타입 변경
- Builder 추가

hek-git committed:
[feature] ReviewMenuEntity 클래스 필드 변경
- Byte 형태 저장을 위한 UUID 타입 변경
- Builder 추가

hek-git committed:
[feature] ReviewController 클래스 기능 추가
- 요청 body를 reviewMenuCreateCommand로 리스트 변환후 reviewCreateCommand 객체 생성
- ReviewCreateResult 객체를 이용한 ReviewCreateResponse 객체 응답

hek-git committed:
[feature] ReviewController 클래스 기능 추가
- 요청 body를 reviewMenuCreateCommand로 리스트 변환후 reviewCreateCommand 객체 생성
- ReviewCreateResult 객체를 이용한 ReviewCreateResponse 객체 응답

hek-git committed:
[feature] ReviewRepository 클래스 변경
- common-domain을 사용하여 Uuid 파싱하도록 변경

hek-git committed:
[feature] ReviewRepository 클래스 변경
- common-domain을 사용하여 Uuid 파싱하도록 변경

hek-git committed:
[feature] ReviewService 클래스 변경
- ReviewCreateResult 객체 응답
- restaurantUuid가 아닌 id를 찾아 넣도록 변경
- menuUuid를 통해 menu의 id를 찾아 넣도록 함
- review 저장 이후 review의 pk를 포함시켜 reviewMenu 저장
- 반환된 id 값들을 가지고 Result 객체 생성 후 반환

hek-git committed:
[feature] ReviewService 클래스 변경
- ReviewCreateResult 객체 응답
- restaurantUuid가 아닌 id를 찾아 넣도록 변경
- menuUuid를 통해 menu의 id를 찾아 넣도록 함
- review 저장 이후 review의 pk를 포함시켜 reviewMenu 저장
- 반환된 id 값들을 가지고 Result 객체 생성 후 반환

jcw1031 committed:
Merge pull request #78 from woowa-techcamp-2024/fix/77-gradle-build-error

[fix] Gradle 빌드 오류 수정

huiseung committed:
[config] csv 파일 gitignore

hellomatia commented on domain/review-domain-rdb/src/main/java/woowa/team4/bff/review/domain/ReviewMenu.java:
```diff
@@ -0,0 +1,19 @@
+@ToString
+public class ReviewMenu {
+
+    private final String reviewMenuUuId;
```
```suggestion
    private final String reviewMenuUuid;
```

hellomatia replied:
r: 여기 대문자 I가 중간에 끼어있네용

hek-git replied:
꼼꼼한 리뷰 감사합니다! 수정했습니다!

hellomatia reviewed (commented):
고생하셨습니다! `menuReview`를 보니까 `reviewId`로 가지고 있느데, 해당 `review`의 content는 null인가요?

huiseung committed:
[refactor] aop 이동

huiseung committed:
[feature] dummy document bulk insert api

huiseung committed:
[feature] mysql like

huiseung committed:
[refacotr] id list 기반으로 검색 목록 찾는 쿼리 시간 측정

huiseung committed:
[feature] index 설정 수정

huiseung committed:
[config] local 시작시 db 초기화 하지 않도록 수정

huiseung committed:
[config] ddl-auto update

huiseung committed:
[refactor] System.out.print -> log 변경

huiseung committed:
Merge pull request #80 from woowa-techcamp-2024/fix/75-search-query

[fix] 검색 목록 조회 쿼리 수정

hek-git committed:
[fix] ReviewMenu 도메인 객체 필드명 오타 수정
- reviewMenuUuId -> reviewMenuUuid로 수정
- 관련 사용 코드 변경

hek-git committed:
[feature] 리뷰 통계를 관리하는 ReviewStatisticsEntity 추가
- 식당 id, 리뷰 개수, 별점 평균 column을 갖고있다.

hek-git committed:
[chore] : 리뷰 작성시 이벤트를 발생시키기 위한 패키지 추가
- common-event domain 의존성 추가
- 리뷰 작성시 발생시킬 ReviewCreateEvent 클래스 추가
   - 식당 Id, 별점 수치를 전달

hek-git committed:
[chore] : 리뷰 작성시 이벤트를 발생시키기 위한 의존성 추가
- event 발행 가능 클래스를 포함하고 있는 event-publisher domain 의존성 추가
- reviewEvent 관련 발행 가능 클래스를 포함하고 있는 review-event domain 의존성 추가

hek-git committed:
[feature] : ReviewStatistics 도메인 클래스 추가
- ReviewStatisticsEntity를 만들기 위한 필드를 갖고 있다.

hek-git committed:
[feature] : ReviewStatisticsEntity를 처리할 수 있는 JPARepository 클래스 추가
- JPARepository 상속
- 식당 Id로 ReviewStatisticsEntity를 찾는 findByRestaurantId 추상 메서드 명세

hek-git committed:
[feature] : ReviewStatistics 도메인 객체를 처리할 수 있는 ReviewStaticsRepository 클래스 추가
- 식당 Id를 이용하여 해당 식당의 리뷰 통계를 조회하는 findByRestaurantId 메서드 구현
- Entity를 저장하는 save 메서드 구현
- Domain -> Entity 변환 메서드 구현
- Entity -> Domain 변환 메서드 구현

hek-git committed:
[feature] : ReviewStatistics 도메인 객체를 처리할 수 있는 ReviewStaticsRepository 클래스 추가
- reviewStatistics 처리를 위한 Repository 의존성 주입
- 이벤트 발행을 위한 publisher 객체 의존성 주입
- ReviewStatistics 리뷰 통계 조회 후 받은 이벤트를 이용하여 update
- ReviewStatistics가 없다면 첫 리뷰이므로 ReviewStatistics 객체를 만들어 저장

hek-git committed:
Merge remote-tracking branch 'origin/feature/67-improve-review-create-function' into feature/67-improve-review-create-function

# Conflicts:
#	domain/review-domain-rdb/src/main/java/woowa/team4/bff/review/domain/ReviewMenu.java
#	domain/review-domain-rdb/src/main/java/woowa/team4/bff/review/repository/ReviewMenuRepository.java
#	domain/review-domain-rdb/src/main/java/woowa/team4/bff/review/service/ReviewService.java
